postgres.go

```diff
@@ -0,0 +1,245 @@
+package dbinitializer
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/docker/go-connections/nat"
+	"github.com/go-playground/errors/v5"
+	shopspring "github.com/jackc/pgx-shopspring-decimal"
+	"github.com/jackc/pgx/v5"
+	"github.com/jackc/pgx/v5/pgxpool"
+
+	_ "github.com/golang-migrate/migrate/v4/database/postgres" // database driver for the migrate package
+	_ "github.com/golang-migrate/migrate/v4/source/file"       // up/down script file source driver for the migrate package
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+)
+
+const (
+	defaultPostgresHost     = "localhost"
+	defaultPostgresPort     = "5432"
+	defaultPostgresDatabase = "postgres"
+)
+
+// PostgresContainer represents a docker container running a postgres instance.
+type PostgresContainer struct {
+	testcontainers.Container
+	host                 string
+	port                 nat.Port
+	superUsername        string
+	unprivilegedUsername string
+	password             string
+	defaultDatabase      string
+
+	sMu                  sync.Mutex
+	superUserConnections map[string]*pgxpool.Pool
+
+	muReplacementCount sync.Mutex
+	replacementCount   int
+}
+
+// NewPostgresContainer returns a new PostgresContainer ready to use with postgres.
+func NewPostgresContainer(ctx context.Context) (*PostgresContainer, error) {
+	pg, err := initPostgresContainer(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	if err := pg.addUnprivilegedUser(ctx); err != nil {
+		return nil, err
+	}
+
+	return pg, nil
+}
+
+// initPostgresContainer returns a PostgresContainer which represents a newly started docker container running postgres.
+func initPostgresContainer(ctx context.Context) (*PostgresContainer, error) {
+	password := "password"
+
+	req := testcontainers.ContainerRequest{
+		Image:        "postgres:latest",
```

Comment:
Do we consider any need to use a certain version of postgres? Or is latest always fine?

Reply:
That is a good idea! I would prefer to that work on a separate PR though so I will add an issue.